test/unit/block/shared/src/BlockSharedMemSt.cpp

```diff
@@ -39,6 +39,32 @@ BOOST_AUTO_TEST_SUITE(blockSharedMemSt)
 //#############################################################################
 //!
 //#############################################################################
+template<
+    typename TType,
+    size_t TSize>
+struct Array {
+    TType m_data[TSize];
```

Comment:
This should be named `data` to be more equeal to a C++11 `std::Array` [link](http://de.cppreference.com/w/cpp/container/array)
This also avoid not nice operations like `&array[0]` to get the pointer of the array.